pycid/core/macid.py

```diff
@@ -2,40 +2,60 @@
 
 import copy
 import itertools
-from typing import Dict, Iterable, List, Optional, Tuple
+from collections import defaultdict
+from functools import partial, update_wrapper
+from typing import Any, Callable, Dict, Hashable, Iterable, KeysView, List, Mapping, Optional, Set, Tuple, Union
+from warnings import warn
 
 import networkx as nx
+import pygambit
 from pgmpy.factors.discrete import TabularCPD
 
 from pycid.core.cpd import DecisionDomain, StochasticFunctionCPD
 from pycid.core.macid_base import MACIDBase
 from pycid.core.relevance_graph import CondensedRelevanceGraph
 
+Outcome = Any
+
 
 class MACID(MACIDBase):
     """A Multi-Agent Causal Influence Diagram"""
 
-    def get_all_pure_ne(self) -> List[List[StochasticFunctionCPD]]:
+    def get_all_ne(self, solver: Optional[str] = "enumpure") -> List[List[StochasticFunctionCPD]]:
         """
-        Return a list of all pure Nash equilbiria in the MACID.
+        Return a list of Nash equilbiria in the MACID.
```

Comment:
add that the default value for `solver' is "enumpure" - which is the pyGambit method for finding all pure NE